crates/sui-framework/packages/sui-framework/tests/test_scenario_tests.move

```diff
@@ -472,6 +472,82 @@ module sui::test_scenario_tests {
         scenario.end();
     }
 
+    // Make sure that we properly handle the case where we receive an object
+    // when dealing with allocated tickets.
```

Comment:
nit: doc comment? Also could you describe what happens when we don't properly handle that case, for posterity?